learnosity_sdk/request.py

```diff
@@ -219,3 +222,62 @@ def set_service_options(self):
     def hash_list(self, l):
         "Hash a list by concatenating values with an underscore"
         return hashlib.sha256("_".join(l).encode('utf-8')).hexdigest()
+
+
+class DataApi(object):
+
+    def request(self, endpoint, security_packet,
+                secret, request_packet, action):
+        """Make a single request to Data API"""
+        init = Init('data', security_packet, secret, request_packet, action)
+        return requests.post(endpoint, data=init.generate())
+
+    def request_recursive(self, endpoint, security_packet,
+                          secret, request_packet=None,
+                          action=None):
+        """Return a list of all results from a request to Data API"""
+        all_data = []
+        for data in self.request_iter(endpoint, security_packet,
+                                      secret, request_packet,
+                                      action):
+            all_data.extend(data['data'])
+        return all_data
+
+    def request_iter(self, endpoint, security_packet,
+                     secret, request_packet=None,
+                     action=None):
+        """
+        Iterate over the results of a query to data api
+
+        A single page of results is yielded each time.
+        """
+
+        # just in case the security_packet or request_packet
+        # are modified between yields
+        security_packet = copy.deepcopy(security_packet)
+        request_packet = copy.deepcopy(request_packet)
+        while True:
```

Comment:
I personally thought the break to be more elegant than having the flag, but I can change it.

Reply:
Yeah, I would prefer to have an explicit termination condition. `break`s are always confusing me, I'm never sure which structure they exit from, and it's not clear from the `while True` that it can actually terminate. Nor can you do a text search for the name of the termination variable to see how it's set.